src/plugins/data_views/server/plugin.ts

```diff
@@ -42,8 +42,9 @@ export class DataViewsServerPlugin
   ) {
     core.savedObjects.registerType(dataViewSavedObjectType);
     core.capabilities.registerProvider(capabilitiesProvider);
+    const dataViewRestCounter = usageCollection?.createUsageCounter('dataViewsRestApi');
 
-    registerRoutes(core.http, core.getStartServices);
+    registerRoutes(core.http, core.getStartServices, dataViewRestCounter);
 
```

Comment:
@mattkime what do you think of accessing the path pattern at the `http` service level? Dreaming here:

```ts
    http.registerOnPreResponse((req, res, t) => {
      //                                                                      t.matchedRoute does not exist
      usageCollection?.incrementCounter({ counterName: `${req.route.method} ${t.matchedRoute.path}` });
      return t.next();
    });
```

I'm not sure how common this need is (or what other things that might address). If you agree though perhaps we can suggest it to core. I may also be misunderstanding when `onPreResponse` is occurring.

Reply:
@jloleysens thats an interesting idea - I'll talk to core about it